jgrapht-core/src/main/java/org/jgrapht/intervalgraph/interval/Interval.java

```diff
@@ -118,21 +133,87 @@ public int compareTo(Interval<T> o) {
     }
 
     @Override
-    public String toString() {
+    public String toString()
+    {
         return "Interval[" + start + ", " + end + "]";
     }
 
     @Override
-    public boolean equals(Object o) {
-        if (this == o) return true;
-        if (o == null || getClass() != o.getClass()) return false;
+    public boolean equals(Object o)
+    {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
+            return false;
         Interval<?> interval = (Interval<?>) o;
-        return Objects.equals(start, interval.start) &&
-                Objects.equals(end, interval.end);
+        return Objects.equals(start, interval.start) && Objects.equals(end, interval.end);
     }
 
     @Override
-    public int hashCode() {
+    public int hashCode()
+    {
         return Objects.hash(start, end);
     }
+
+    /**
+     * getter for a Comparator, which only compares the starting points
+     * 
+     * @param <T> the value of the interval
+     * @return a starting point comparator
+     */
+    public static <T extends Comparable<T>> StartingComparator<T> getStartingComparator()
+    {
+        return new StartingComparator<T>();
+    }
+
+    /**
+     * getter for a Comparator, which only compares the ending points
+     * 
+     * @param <T> the value of the interval
+     * @return a ending point comparator
+     */
+    public static <T extends Comparable<T>> EndingComparator<T> getEndingComparator()
+    {
+        return new EndingComparator<T>();
```

Comment:
can even directly return an anonymous class instead of declare another inner class.